op-service/sources/types.go

```diff
@@ -183,31 +196,9 @@ func (block *rpcBlock) verify() error {
 	if computed := block.computeBlockHash(); computed != block.Hash {
 		return fmt.Errorf("failed to verify block hash: computed %s but RPC said %s", computed, block.Hash)
 	}
-	for i, tx := range block.Transactions {
-		if tx == nil {
-			return fmt.Errorf("block tx %d is null", i)
-		}
-	}
```

Comment:
I suspect we want to keep this to avoid potentially dereferencing a nil and crashing if the RPC provider gives us bad data.